Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -1108,11 +1155,6 @@ private static bool CheckServiceMatch(Type interfaceType, string serviceName, Ty
         /// <returns></returns>
         private static bool CanGetService(Type interfaceType)
         {
-            if (isApplicationQuitting)
```

Comment:
What's the benefit of removing this?

Reply:
keeping it prevents us from getting the service on shutdown.

Reply:
If this can be called during runtime (and shoudl probably continue to have this restriction) maybe it makes sense to keep this behavior conditional on some method parameter.

---
In reply to: [288757284](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4619#discussion_r288757284) [](ancestors = 288757284)

Reply:
@wiwei, agree that at this point in the release, it is prudent to be mindful of unexpected changes. will add an optional param

Reply:
added class global internalShutdown variable. also removed unneeded ExecuteOnAllServices all